Source/SwiftLintFramework/Extensions/Configuration+Parsing.swift

```diff
@@ -202,4 +203,15 @@ extension Configuration {
             }
         }
     }
+
+    private static func warnAboutMisplacedAnalyzerRules(optInRules: [String], ruleList: RuleList) {
+        let analyzerRules = ruleList.list
+            .filter { $0.value.self is AnalyzerRule.Type }
+            .map(\.key)
+        let invalidKeys = Set(analyzerRules).intersection(optInRules)
+        if invalidKeys.isNotEmpty {
+            queuedPrintError("warning: Analyzer rules \(invalidKeys) are opt-in by default and should be " +
```

Comment:
Can this sort and pretty-print the rule IDs? right now it's a `Set` so the order could change every time this is run, and the way it's embedded in a sentence might not look good. Maybe a dashed list at the end?

> warning: Analyzer rules must be listed in the 'analyzer_rules' configuration section:
> - explicit_self
> - unused_declaration
> - unused_import

Reply:
The set is now sorted. Like in the other warnings, for every misplaced rule, a separate warning is shown.